cli/tests/integrations/test_marathon_pod.py

```diff
@@ -144,8 +143,37 @@ def _assert_pod_json(expected_pod, actual_pod):
     assert len(actual_containers) == len(expected_containers)
 
 
-def _assert_pod_list_table(stdout):
-    assert_command(_POD_LIST_CMD, returncode=0, stdout=stdout, stderr=b'')
+def _assert_pod_list_table():
+    for attempt in range(10):
+        returncode, stdout, stderr = exec_command(_POD_LIST_CMD)
+        stdout_lines = stdout.decode('utf-8').split('\n')
+
+        expected_instances = ['2', '3', '1']
+        actual_instances = [stdout_lines[i].split()[1] for i in [1, 4, 6]]
+        if actual_instances == expected_instances:
+            break
```

Comment:
This is very hacky. Lets have a wait_for_instances like we wait_for_deployments.